lib/autoproj/cli/main.rb

```diff
@@ -3,13 +3,20 @@
 require 'autoproj/cli/main_test'
 require 'autoproj/cli/main_plugin'
 require 'autoproj/cli/main_global'
+require 'autoproj/cli'
 require 'autoproj/reporter'
 
 module Autoproj
     module CLI
         def self.basic_setup
             Encoding.default_internal = Encoding::UTF_8
             Encoding.default_external = Encoding::UTF_8
+
+            argv = ARGV.find_all { |arg| arg != "--no-plugins" }
```

Comment:
why can't we have this integrated in the Thor interface?

Reply:
Because plugins extend the thor interface (they add commands & subcommands). So, need to be loaded before you start executing Thor